### PR TITLE
Feat/st 337

### PR DESCRIPTION
**Pull request by muradalamv1**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/ciccase/model/CicCase.java [hunk 1]
```diff
@@ -36,7 +35,7 @@ public class CicCase {
     @CCD(
         label = "Case Subcategory",
         access = {DefaultAccess.class, CaseworkerWithCAAAccess.class},
-        typeOverride = MultiSelectList,
+        typeOverride = FixedRadioList,
         typeParameterOverride = "CaseSubcategory"
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/divorce/ciccase/model/CicCase.java:43]: Are these classnames?

File: src/main/java/uk/gov/hmcts/divorce/ciccase/model/RepresentativeCIC.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@Getter
+@AllArgsConstructor
+public enum RepresentativeCIC implements HasLabel {
+    @JsonProperty("RepresentativeCIC")
+    REPRESENTATIVE("Representative");
+    private final String label;
+    public boolean isRepresentativeCIC() {
+        return REPRESENTATIVE.name().equalsIgnoreCase(this.name());
+    }
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/divorce/ciccase/model/RepresentativeCIC.java:16]: If REPRESENTATIVE is an enum, use the == operator to check equality, not String equality.
== is null safe, .equals is not (Though you're not going to get an NPE here, it's just better practice)

File: src/main/java/uk/gov/hmcts/divorce/common/event/CreateTestCase.java [hunk 3]
```diff
@@ -69,7 +69,36 @@ public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
...
+            .page("applicantDetailsObject")
+            .label("applicantDetailsObject","Who is the subject of this case?\r\n" + "\r\nCase record for [DRAFT]")
+            .complex(CaseData::getCicCase)
+            .mandatory(CicCase::getFullName)
+            .optional(CicCase::getAddress)
+            .optional(CicCase::getPhoneNumber)
+            .optional(CicCase::getEmail)
+            .mandatoryWithLabel(CicCase::getDateOfBirth,"")
+            .mandatoryWithLabel(CicCase::getContactDetailsPrefrence,"")
+            .done();
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/divorce/common/event/CreateTestCase.java:101]: This is very long and poses some readability issues, is it possible to chunk it into logical portions, and make it a little more readable?

**Review by osheaf01 — APPROVED** [review]
LGTM.
My comments are mild style suggestions rather than faults.

File: src/main/java/uk/gov/hmcts/divorce/ciccase/model/SubjectCIC.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+public enum SubjectCIC implements HasLabel {
+    @JsonProperty("SubjectCIC")
+    SUBJECT("Subject");
+
+
+    private final String label;
+
+    public boolean isSubject() {
+        return SUBJECT.name().equalsIgnoreCase(this.name());
+    }
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/divorce/ciccase/model/SubjectCIC.java:19]: Same comment re enums as above.

**Review by jaberelfasi — APPROVED** [review]

**Pull request merged**